cluster/vagrant/provision-master.sh

```diff
@@ -34,6 +35,9 @@ if [[ "$(grep 'VERSION_ID' /etc/os-release)" =~ ^VERSION_ID=23 ]]; then
 fi
 
 NETWORK_IF_NAME=`echo ${if_to_edit} | awk -F- '{ print $3 }'`
+if [[ -z "$NETWORK_IF_NAME" ]]; then
+  NETWORK_IF_NAME="eth0"
```

Comment:
This has been a pain point to get right, so a comment or some text to help me explain would be useful.

Is vsphere not adding `VAGRANT-BEGIN` comments at the start of the network-script or is it that in your case if_to_edit is empty because it does not have access to the NETWORK_CONF_PATH?  Keeping a bread-crumb here that says this was was needed for vsphere support because will help it not later get removed...

I also wonder if we can have a variable in config-default.sh for the fallback NETWORK_IF_NAME in case other providers run into a difference.